src/compiler/crystal/macros/methods.cr

```diff
@@ -699,7 +699,7 @@ module Crystal
         end
 
         if value
-          NumberLiteral.new(value)
+          NumberLiteral.new(value.to_s, :i32)
```

Comment:
a few line above the value is converted to an `Int64` (`@value.to_i64?`), it feels weird to 'cast' it to `:i32` here, is it intentional ?

Reply:
It is intentional as far as this prevents codegen to fail. 🤷‍♂️ It's only a workaround, though.
In fact, this isn't even a change: it implicitly was `:i32` before, now it's just explicit.

But yes, it should probably be adjusted, however we need to fix the codegen bug first.